src/include/devcomm.h

```diff
@@ -195,6 +196,9 @@ struct scclAlgorithm {
   // declaration for scratchBuffer. This is only to be accessed by the host
   size_t scratchBufferSize;
   void* scratchBuffer;
+  //Reduction Operator. If the algorithm performs reduction it will specify the reduction operator.
+  //If the algorithm do not perform reduction, its reduction operator is considered as ncclSum.
+  ncclRedOp_t redOp;
```

Comment:
yeah, still not sure why we need this. FUNC will apply the same reduction operation all across, right?